checker/src/main/java/org/checkerframework/checker/signedness/jdk.astub

```diff
@@ -3313,29 +3313,94 @@ class DigestOutputStream {
-    public static @PolySigned char[] copyOfRange(@PolySigned char[] original, int from, int to);
-
-    public static @PolySigned byte[] copyOf(@PolySigned byte[] original, int newLength);
-    public static @PolySigned short[] copyOf(@PolySigned short[] original, int newLength);
-    public static @PolySigned int[] copyOf(@PolySigned int[] original, int newLength);
-    public static @PolySigned long[] copyOf(@PolySigned long[] original, int newLength);
-    public static @PolySigned char[] copyOf(@PolySigned char[] original, int newLength);
+    public static int hashCode(@PolySigned byte[] a);
+    public static int hashCode(@PolySigned short[] a);
+    public static int hashCode(@PolySigned int[] a);
+    public static int hashCode(@PolySigned char[] a);
+    public static int hashCode(@PolySigned long[] a);
+
+    public static void parallelPrefix(@PolySigned int[] array, IntBinaryOperator op);
+    public static void parallelPrefix(@PolySigned long[] array, LongBinaryOperator op);
+    public static void parallelPrefix(@PolySigned int[] array, int fromIndex, int toIndex, IntBinaryOperator op);
+    public static void parallelPrefix(@PolySigned long[] array, int fromIndex, int toIndex, LongBinaryOperator op);
+
+    public static void parallelSetAll(@PolySigned int[] array, IntUnaryOperator generator);
+    public static void parallelSetAll(@PolySigned long[] array, IntToLongFunction generator);
+
+    public static void parallelSort(@PolySigned byte[] a);
+    public static void parallelSort(@PolySigned short[] a);
+    public static void parallelSort(@PolySigned int[] a);
+    public static void parallelSort(@PolySigned long[] a);
+    public static void parallelSort(@PolySigned char[] a);
+    public static void parallelSort(@PolySigned byte[] a, int fromIndex, int toIndex);
+    public static void parallelSort(@PolySigned short[] a, int fromIndex, int toIndex);
+    public static void parallelSort(@PolySigned int[] a, int fromIndex, int toIndex);
+    public static void parallelSort(@PolySigned long[] a, int fromIndex, int toIndex);
+    public static void parallelSort(@PolySigned char[] a, int fromIndex, int toIndex);
+
+    public static void sort(@PolySigned byte[] a);
+    public static void sort(@PolySigned short[] a);
+    public static void sort(@PolySigned int[] a);
+    public static void sort(@PolySigned long[] a);
+    public static void sort(@PolySigned char[] a);
+    public static void sort(@PolySigned byte[] a, int fromIndex, int toIndex);
+    public static void sort(@PolySigned short[] a, int fromIndex, int toIndex);
+    public static void sort(@PolySigned int[] a, int fromIndex, int toIndex);
+    public static void sort(@PolySigned long[] a, int fromIndex, int toIndex);
+    public static void sort(@PolySigned char[] a, int fromIndex, int toIndex);
+
+    public static void setAll(@PolySigned int[] array, IntUnaryOperator generator);
```

Comment:
These annotations look unsound to me.  They permit the use of any generator with any array.
I think it's better to require the user to suppress a warning, rather than to permit unsafe operations.
You could write a comment indicating why they are not annotated.